SRM deployment fails if more than four hosts are present in a cluster

### Code of Conduct

- [X] I have read and agree to the [Code of Conduct](https://github.com/vmware/power-validated-solutions-for-cloud-foundation/blob/main/CODE_OF_CONDUCT.md).
- [X] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [X] Do not leave "+1" or other comments that do not add relevant information or questions.
- [X] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### VMware Cloud Foundation

5.2

### Module Version

2.11.0.1057

### PowerShell Version

7.4.4

### PowerCLI Version

13.3

### PowerVCF Version

2.4.1.1000

### Guest Operating System

Windows Server 2019

### Environment Details

_No response_

### Description

If more than four hosts are present in a cluster, the PDR deployment of SRM fails with

 [ERROR] The number of IP addresses supplied do not match the number if ESXi hosts in Workload Domain ($DOMAIN): PRE_VALIDATION_FAILED

Please handle more than four hosts and correct the typo "of ESXi hosts" rather than "is ESXi hosts"


### Error or Debug Output

 [ERROR] The number of IP addresses supplied do not match the number if ESXi hosts in Workload Domain ($DOMAIN): PRE_VALIDATION_FAILED


### Expected Behavior

Generate a JSON with replication IPs equal to the host count and validate this as part of the pre-check.

### Actual Behavior

Fails with  [ERROR] The number of IP addresses supplied do not match the number if ESXi hosts in Workload Domain ($DOMAIN): PRE_VALIDATION_FAILED


### Steps to Reproduce

Deploy a WLD with more than four hosts

### Log Fragments and Files

_No response_

### Screenshots

_No response_

### References

_No response_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -13,6 +13,7 @@
 - Enhanced `Install-vSphereReplicationManager` cmdlet to require a portgroup as an input.
 - Enhanced `Install-SiteRecoveryManager` cmdlet to require a portgroup as an input.
 - Enhanced `Invoke-PdrDeployment` cmdlet to pass the new portgroup parameter to `Install-vSphereReplicationManager` and `Install-SiteRecoveryManager`.
+- Enhanced `Export-PdrJsonSpec` cmdlet to extract the range of replication IPs based on the populated Planning and Preparation workbook.
 
 ## v2.11.0
 

--- a/PowerValidatedSolutions.psd1
+++ b/PowerValidatedSolutions.psd1
@@ -11,7 +11,7 @@
     RootModule = 'PowerValidatedSolutions.psm1'
 
     # Version number of this module.
-    ModuleVersion = '2.11.1.1005'
+    ModuleVersion = '2.11.1.1006'
 
     # ID used to uniquely identify this module
     GUID              = 'a6dfed7b-65d2-4da2-bdcc-7f3d3df9b75d'

--- a/PowerValidatedSolutions.psm1
+++ b/PowerValidatedSolutions.psm1
@@ -2581,7 +2581,7 @@ Function Export-PdrJsonSpec {
                     'replicationGateway'       = if ($null -eq $pnpProtectedWorkbook.Workbook.Names["mgmt_vrms_gateway_ip"].Value) { $pnpProtectedWorkbook.Workbook.Names["mgmt_az1_vrms_gateway_ip"].Value } else { $pnpProtectedWorkbook.Workbook.Names["mgmt_vrms_gateway_ip"].Value }
                     'replicationNetmask'       = $pnpProtectedWorkbook.Workbook.Names["mgmt_vrms_mask"].Value
                     'remoteReplicationNetwork' = $pnpProtectedWorkbook.Workbook.Names["mgmt_vrms_recovery_replication_cidr"].Value
-                    'replicationIpAddresses'   = @("$($pnpProtectedWorkbook.Workbook.Names["mgmt_az1_host1_vrms_ip"].Value)", "$($pnpProtectedWorkbook.Workbook.Names["mgmt_az1_host2_vrms_ip"].Value)", "$($pnpProtectedWorkbook.Workbook.Names["mgmt_az1_host3_vrms_ip"].Value)", "$($pnpProtectedWorkbook.Workbook.Names["mgmt_az1_host4_vrms_ip"].Value)")
+                    'replicationIpAddresses'   = ($pnpProtectedWorkbook.Workbook.Names["mgmt_az1_host_vrms_ips"].Value) | Where-Object { $_ -ne "Value Missing" }
                     'vmFolderSrm'              = $pnpProtectedWorkbook.Workbook.Names["mgmt_srm_vm_folder"].Value
                     'srmVmName'                = $pnpProtectedWorkbook.Workbook.Names["mgmt_srm_hostname"].Value
                     'srmFqdn'                  = $pnpProtectedWorkbook.Workbook.Names["mgmt_srm_fqdn"].Value
@@ -2642,7 +2642,7 @@ Function Export-PdrJsonSpec {
                     'replicationGateway'       = if ($null -eq $pnpRecoveryWorkbook.Workbook.Names["mgmt_vrms_gateway_ip"].Value) { $pnpRecoveryWorkbook.Workbook.Names["mgmt_az1_vrms_gateway_ip"].Value } else { $pnpRecoveryWorkbook.Workbook.Names["mgmt_vrms_gateway_ip"].Value }
                     'replicationNetmask'       = $pnpRecoveryWorkbook.Workbook.Names["mgmt_vrms_mask"].Value
                     'remoteReplicationNetwork' = $pnpRecoveryWorkbook.Workbook.Names["mgmt_vrms_recovery_replication_cidr"].Value
-                    'replicationIpAddresses'   = @("$($pnpRecoveryWorkbook.Workbook.Names["mgmt_az1_host1_vrms_ip"].Value)", "$($pnpRecoveryWorkbook.Workbook.Names["mgmt_az1_host2_vrms_ip"].Value)", "$($pnpRecoveryWorkbook.Workbook.Names["mgmt_az1_host3_vrms_ip"].Value)", "$($pnpRecoveryWorkbook.Workbook.Names["mgmt_az1_host4_vrms_ip"].Value)")
+                    'replicationIpAddresses'   = ($pnpProtectedWorkbook.Workbook.Names["mgmt_az1_host_vrms_ips"].Value) | Where-Object { $_ -ne "Value Missing" }
                     'vmFolderSrm'              = $pnpRecoveryWorkbook.Workbook.Names["mgmt_srm_vm_folder"].Value
                     'srmVmName'                = $pnpRecoveryWorkbook.Workbook.Names["mgmt_srm_hostname"].Value
                     'srmFqdn'                  = $pnpRecoveryWorkbook.Workbook.Names["mgmt_srm_fqdn"].Value

